### PR TITLE
[Doc] Move public TensorFlow runtime APIs to new file.

### PR DESCRIPTION
**Pull request by dan-zheng**
Move public APIs from `CompilerRuntime.swift` to new file `Execution.swift`.
This clearly separates user-exposed public functions from compiler internals.

This separation also benefits `jazzy` API doc generation: `jazzy` can simply
skip doc generation for the entire `CompilerRuntime.swift` file. Otherwise,
per-declaration `/// :nodoc:` annotations would be necessary for skipping
documentation.

### CODE REVIEW

**Review by rxwei — APPROVED** [review]
Thanks!

File: stdlib/public/TensorFlow/Execution.swift [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+//
+// Copyright (c) 2014 - 2017 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+//
+// This file defines public APIs regarding TensorFlow runtime support.
```
Comment (mhong) [inline stdlib/public/TensorFlow/Execution.swift]: does `CompilerRuntime.swift` have a corresponding message that it only contains internal details?

**Review by mhong — APPROVED** [review]
nice!

File: stdlib/public/TensorFlow/Execution.swift (hunk 1 above)
Comment (dan-zheng) [reply stdlib/public/TensorFlow/Execution.swift]: Thanks for the great point! Done in https://github.com/apple/swift/pull/23032/commits/c855053745ade9eb6862a999fc88972090a33678.

**Comment by dan-zheng** [comment]
@swift-ci Please test tensorflow

**Pull request merged**